extensions/BugModal/lib/WebService.pm

```diff
@@ -62,6 +81,26 @@ sub rest_resources {
     ]
 }
 
+sub products {
+    my $user = Bugzilla->user;
+    my @products = @{ $user->get_enterable_products };
+
+    @products = map { { name => $_->name } } @products;
+    return {
+        products     => \@products,
+    };
+}
+
+sub components {
+    my ($self, $params) = @_;
+    my $product = Bugzilla::Product->check({ name => $params->{product_name}, cache => 1 });
+    
+    my @components = map { { name => $_->name } } @{ $product->components };
```

Comment:
I admit I don't quite follow this specific comment, sorry. In the other place I was noting that it was weird how the two data structures were constructed in completely different ways, when the code flow itself was the sole possible way to guess at the structure.